composables/use-draggable-scroll.ts

```diff
@@ -0,0 +1,148 @@
+import type { Ref } from 'vue'
+import { getHtmlElement } from '~/utils/ref/get-html-element'
+
+interface Point {
+    x: number
+    y: number
+}
+
+interface UseDraggableScrollOptions {
+    element: Ref<HTMLElement | null> | HTMLElement
+}
+
+// Amount of pixels for detecting if the element is being dragged or just clicked.
+const MIN_DRAG_AMOUNT = 6
+
+export function useDraggableScroll(options: UseDraggableScrollOptions) {
+    const isDragging = ref(false)
+    const hasOverflow = ref(false)
+
+    let listen = false
+    let dragged = false
+    let dragAmount: Point = { x: 0, y: 0 }
+    let resizeObserver: ResizeObserver | null = null
+
+    function checkAvailability(direction = 'x') {
+        const el = getHtmlElement(options.element)
+        if (!el) return false
+
+        const lastChildren = el.children[el.children.length - 1] as HTMLElement
+        const boundingKey = direction === 'x' ? 'right' : 'bottom'
+
+        return el.getBoundingClientRect()[boundingKey] < lastChildren.getBoundingClientRect()[boundingKey]
+    }
+
+    function onMouseDown() {
+        removeListeners()
```

Comment:
there is no need to call this function here, event listener are added at the end of this function but are removed in the mouseUp function, it is not listened here